ui/apps/platform/src/Containers/Integrations/IntegrationsListPage/DownloadCAConfigBundle.tsx

```diff
@@ -1,13 +1,15 @@
 import React, { ReactElement, useState } from 'react';
+import { useDispatch } from 'react-redux';
 import { Button, Tooltip } from '@patternfly/react-core';
 import FileSaver from 'file-saver';
 
+import { actions } from 'reducers/notifications';
 import { fetchCAConfig } from 'services/ClustersService';
-import useNotifications from 'hooks/useNotifications';
+import { getAxiosErrorMessage } from 'utils/responseErrorUtils';
 
 function DownloadCAConfigBundle(): ReactElement {
     const [downloadingCAConfig, setDownloadingCAConfig] = useState<boolean>(false);
-    const addNotification = useNotifications();
+    const dispatch = useDispatch();
```

Comment:
Because there are 30 occurrences of `dispatch` with action creator function compared to only this one call via `useNotifications` hook, it seemed worthwhile to take a step back before we decide future direction.

As mentioned in **Residue** section discusses there are also 6 occurrences of `useToasts` hook.

Off the topic of pro and con to retain or replace the hook, it had a subtle problem with `setTimeout` call which threw an error when the `catch` block ran:

```js
setTimeout(dispatch(actions.removeOldestNotification), 5000);
```

Reply:
Aha, thinking about this some more: a fluent solution for future follow up is thunk in reducer.

@bradr5 has replaced fetch side-effects in classic sagas with thunks in reducers:
* https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/reducers/featureFlags.js#L14-L28
* https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/reducers/publicConfig.ts#L14-L28
* https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/reducers/telemetryConfig.js#L34

After we reach a consensus about notifications and toasts, instead of export add and remove action creator functions, reducer can export thunk function to add, and (maybe optional) then remove notification via encapsulated setTimeout side-effect (with possible optional duration).

There seems to be a difference between older notifications and newer toasts:
* notifications remove oldest notification at expiry time means the current when there is (usually) only one.
* toasts removes the current by key when it reaches its expiry time.

The toasts interpretation of default removal seems more predictable.